scripts/generate_changelog_local.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/sh
+
+# MIT License
+#
+# Copyright (c) 2024 Stephan Michard
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Based on https://github.com/smichard/conventional_changelog
+
+# Exit on error
+set -e
+
+# Error handling
+trap 'echo "An error occurred at line $LINENO. Exiting."' ERR
+
+# Path to the Git repository (current directory)
+REPO_DIR="."
+CHANGELOG_FILE="$REPO_DIR/CHANGELOG.md"
+GITHUB_REPO_URL=$(git remote get-url origin 2>/dev/null | sed 's/\.git$//')
+if [ -z "$GITHUB_REPO_URL" ]; then
+	GITHUB_REPO_URL=0
+fi
+
+echo "Starting changelog generation script..."
+echo "Repository:"
+echo $GITHUB_REPO_URL
+# Create or clear the changelog file
+> $CHANGELOG_FILE
+
+# Add the introductory text to the changelog
+#echo "# Changelog" >> $CHANGELOG_FILE
+#echo "" >> $CHANGELOG_FILE
+#echo "All notable changes to this project will be documented in this file." >> $CHANGELOG_FILE
+#echo "" >> $CHANGELOG_FILE
+#echo "The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html) and to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)." >> $CHANGELOG_FILE
+#echo "" >> $CHANGELOG_FILE
+
+# Go to the repository directory
+cd $REPO_DIR
+
+# Define categories
+CATEGORIES="feat fix ci perf docs test chore refactor"
```

Comment:
what about PRs that have both feat and fix components
also what about reworks (like e.g. the sc2 ffa MS display where it is more than a refactor as it significantly changes and overhauls the display) would those just be feats?

Reply:
![image](https://github.com/user-attachments/assets/7d4f5d11-88a4-4be7-a66e-6a6242939576)

Though I merged `build` into `ci` and `style` into `refactor`